tests/store/crud/read/test_read.py

```diff
@@ -1424,6 +1428,59 @@ def test_get_number_of_reads_for_sample_with_some_not_passing_q30_threshold(
     assert number_of_reads < total_sample_reads
 
 
+def test_get_illumina_metrics_entry_by_run_id_sample_internal_id_and_lane(
+    store_with_illumina_sequencing_data: Store,
+    novaseq_x_flow_cell_id: str,
+    seven_canonical_flow_cells_selected_sample_ids: list[list[str]],
+    small_helpers: SmallHelpers,
+):
+    """Test that a Illumina sample sequencing metrics query is filtered by sample, run and lane."""
+    # GIVEN a store with Illumina Sample Sequencing Metrics for each sample in the run directories
+    metrics: Query = store_with_illumina_sequencing_data._get_query(
+        table=IlluminaSampleSequencingMetrics
+    )
+    assert metrics.count() == small_helpers.length_of_nested_list(
+        seven_canonical_flow_cells_selected_sample_ids
+    )
```

Comment:
Shouldn't this be after the # WHEN statement?

Reply:
No, this is the number of sample metrics before filtering. But Im thinking that this assertion maybe is not needed here at all